packages/player/src/components/player.tsx

```diff
@@ -72,46 +96,50 @@ export const selectFormattedRemainingTime = (
   state: StateFrom<typeof videoMachine>,
 ) =>
   formatVideoTime(
-    (state.context.video?.duration ?? 0) -
-      (state.context.video?.currentTime ?? 0),
+    (state.context.videoRef?.current?.duration ?? 0) -
+      (state.context.videoRef?.current?.currentTime ?? 0),
   )
 
 export const selectFormattedDuration = (
```

Comment:
migrated all of the selectors into this central spot and think that the next step is likely to move them into their own file

excellent place to start writing a few tests as well